app/Http/Requests/AdminNewVoucherRequest.php

```diff
@@ -27,10 +27,37 @@ public function rules()
         return  [
             // MUST be present, and exist in table
             'sponsor_id' => 'required|exists:sponsors,id',
+            // MUST be present, and a zero padded number string
+            'start' =>' required|string|regex:/^[0-9]+$/',
+            'end' => 'required|string|regex:/^[0-9]+$/',
             // MUST be present, and integer and between 1 and 99999999
-            'start' => 'required|integer|between:1,99999999',
+            'start-serial' => 'between:0,99999999',
             // MUST be present, and integer, greater/equal to start and between 1 and 99999999
-            'end' => 'required|integer|between:1,99999999|ge_field:start'
+            'end-serial' => 'between:0,99999999|ge_field:start'
+        ];
+    }
+
+    // We need to premangle before validating rules
+    public function prepareForValidation()
+    {
+        // create integer versions to be checked
+        $input = array_filter(
+            $this->all(['start', 'end']),
+            'strlen'
+        );
+        foreach ($input as $key => $value) {
+            $input[$key . "-serial"] = intval($value);
+        }
+        $this->merge($input);
+    }
+
+    // Friendlier messages to reference computed fields to normal ones.
+    public function messages()
+    {
+        return  [
+            'start-serial.between.numeric' => 'The start must be between :min and :max.',
+            'end-serial.between.numeric' => 'The end must be between :min and :max.',
+            'end-serial.ge_field' => 'The end must be greater than or equal to start.'
```

Comment:
This should probably be 'The end must be greater than or equal to the start' or 'End must be greater than or equal to start'

Reply:
Purely copy